File: Java/NumeroPrimo.java
public class NumeroPrimo {
    public static void main(String[] args) {
        int numero = 29; 

        if (isPrimo(numero)) {
            System.out.println(numero + " é um número primo.");
        } else {
            System.out.println(numero + " não é um número primo.");
        }
    }

    private static boolean isPrimo(int n) {
        if (n <= 1) {
            return false;
        }
        if (n == 2) {
            return true;
        }
        if (n % 2 == 0) {
            return false;
        }
        for (int i = 3; i * i <= n; i += 2) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }
}

File: JS/numeroPrimo.js
const numero = 11;

let ePrimo = true;

if (numero <= 1) {
    ePrimo = false;
} else if (numero <= 3) {
    ePrimo = true; 
} else if (numero % 2 === 0 || numero % 3 === 0) {
    ePrimo = false; 
} else {
    
    for (let i = 5; i * i <= numero; i += 6) {
        if (numero % i === 0 || numero % (i + 2) === 0) {
            ePrimo = false;
            break;
        }
    }
}
console.log(ePrimo);
